src/daemon/bitcoind/poller.rs

```diff
@@ -8,6 +8,7 @@ use crate::daemon::{
         },
         BitcoindError,
     },
+    cpfp::{get_cpfp_transaction, CpfpError, CpfpTx},
```

Comment:
In commit "Track and broadcast CPFP transaction when needed", you are going through a lot of hassle managing this cache. But do we need to track the CPFP transactions?

What would be wrong with an approach similar to the way we broadcast the Spend transactions (apart from being bulky):
- On new blocks, iterate through the Spend transactions that are waiting for being broadcast (as we do now)
- Store the ones that are high prio and whose Unvaults are still unconfirmed
- After the loop, create a CPFP spending all the unconfirmed Unvaults (and Spends) that were marked as priority
- Take care to check if there is already a CPFP tx in your mempool, if so increase the fees accordingly to replace it (more absolute fees, an added feerate >1sat/vbyte)

And ta-da: you don't need all the complexity of managing the cache (which currently fails in some cases as you outline in the commit message) and you even get RBF for free. Am i missing something?